detectron2/export/flatten.py

```diff
@@ -307,6 +322,18 @@ def forward(self, *args: torch.Tensor):
                         "cannot flatten to tensors."
                     )
             else:  # schema is valid
+                # During torch.onnx.export(), extra outputs that `Schema` implementations
+                # can generate (e.g. `InstancesSchema`) are dropped from final ONNX graph.
+                # This is OK because ONNX graphs do not need to rebuild original data.
+                if len(flattened_output_tensors) > len(flattened_outputs) and\
+                        self.__is_in_torch_onnx_export:
+                    warnings.warn("PyTorch ONNX export (`torch.onnx.export`) detected!"
+                                  " To prevent extra outputs in the ONNX graph, the original"
+                                  " model output cannot be reconstructed through"
+                                  " `adapter.outputs_schema(flattened_outputs)`."
+                                  " For results evaluation, use `torch.jit.trace` instead.")
+                    flattened_outputs = flattened_outputs[:len(outputs)]
+
                 if self.outputs_schema is None:
```

Comment:
Does it means that the final output will need to be processed to be well formatted and ready to use?